docs/domain/domain-model.gql

```diff
@@ -0,0 +1,165 @@
+scalar Hash
+scalar MsFromPosixEpoch
+scalar Base64Url
+scalar Int64
+scalar BigInt
+scalar Base64UrlTransaction
+scalar Object
+
+type Query {
+  block: Block
+}
+
+type Signature {
+  sig: String
+}
+
+type Block {
+  creationTime: MsFromPosixEpoch
+  parent: Block
+  height: BigInt
+  hash: Hash
+  chainId: Int
+  weight: Base64Url
+  featureFlags: Int
+  epochStart: MsFromPosixEpoch
+  adjacents: [Block]
+  payloadHash: Base64Url # fk BlockPayload
+  chainwebVersion: ChainwebNetwork
+  target: Base64Url
+  nonce: Int64
+
+  payload: BlockPayload
+}
+
+type BlockPayload {
+  transactions: [Transaction] # pact realm
+  minerData: Account
+  transactionsHash: Base64Url
+  outputsHash: Base64Url
+  payloadHash: Base64Url
+  coinbase: CoinbaseTransaction
+}
+
+type CoinbaseTransaction {
+  gas: Int
+  result: CoinbaseTransactionResult
+  reqKey: String
+  logs: String
+  events: [EventData]
+  metaData: undefined
+  continuation: undefined
+  txId: number
+}
+
+type CoinbaseTransactionResult {
+  status: String
+  data: String
+}
+
+type Account {
+  name: String
+  guard: KeySet
+}
+
+type EventData {
+
+}
+
+type Account {
+  name: String
+  guard: KeySet
+}
+
+type KeySet {
+  """
+  "keys-all" | "keys-any" | "keys-2"
+  """
+  predicate: String
+  pubKeys: [String]
+}
+
+enum KeysetPredicateOptions {
+  KeysAllPredicate
+  KeysTwoPredicate
+  KeysAnyPredicate
+}
+
+type Transaction {
+  command: TransactionCommand
+  commandResult: TransactionCommandResult
```

Comment:
Some services may not have the results available (e.g. they are not needed for validation and in the future we decide to make storage of results in nodes optional). So, it probably makes sense to keep results `nullable`.

```suggestion
  command: TransactionCommand!
  commandResult: TransactionCommandResult
```